test/e2e/ownerrefs_finalizers_test.go

```diff
@@ -127,6 +128,24 @@ var _ = Describe("Ensure OwnerReferences and Finalizers are resilient [vcsim] [s
 						forceCancelFunc()
 					}
 
+					// This check ensures that the rollout to the machines finished before
+					// checking resource version stability.
+					Eventually(func() error {
+						machineList := &clusterv1.MachineList{}
+						if err := proxy.GetClient().List(ctx, machineList, ctrlclient.InNamespace(namespace)); err != nil {
+							return errors.Wrap(err, "list machines")
+						}
+
+						for i := range machineList.Items {
+							machine := &machineList.Items[i]
```

Comment:
Doesn't really matter, but we can start using regular "iteration variables" since Go 1.22 :)